src/Thentos/Transaction/TH.hs

```diff
@@ -7,36 +10,243 @@ module Thentos.Transaction.TH
     )
 where
 
-import Control.Applicative ((<$>))
+import Control.Applicative (pure, (<$>), (<*>))
 import Control.Monad (replicateM)
-import Data.Acid (Update, Query, makeAcidic)
+import Data.Acid (Update, Query, UpdateEvent, QueryEvent)
+import Data.Acid.Advanced (IsAcidic(..), Event(QueryEvent, UpdateEvent), Method, MethodResult, MethodState)
+import Data.Char (toUpper)
+import Data.Data (Typeable)
+import Data.SafeCopy (SafeCopy(putCopy, getCopy), safeGet, safePut, contain)
 import Language.Haskell.TH
 import Text.Show.Pretty (ppShow)
 
 import Thentos.Transaction.Core (ThentosUpdate, ThentosQuery)
-import Thentos.Types (DB, ThentosError)
+import Thentos.Types (ThentosError, Extends)
 
-data ThentosTransactionType = ThentosQ | ThentosU
 
-makeThentosAcidicPhase1 :: [Name] -> Q [Dec]
-makeThentosAcidicPhase1 names = concat <$> mapM processTransaction names
+data TransactionType = TransQuery | TransUpdate
 
-makeThentosAcidicPhase2 :: Name -> [Name] -> Q [Dec]
-makeThentosAcidicPhase2 stateName eventNames =
-    makeAcidic stateName $ map dropPrefix eventNames
+makeThentosAcidicPhase1 :: Name -> [Name] -> Q [Dec]
+makeThentosAcidicPhase1 dbTypeName names =
+    concat <$> mapM (processTransaction $ ConT dbTypeName) names
 
-processTransaction :: Name -> Q [Dec]
-processTransaction functionName = do
+{-
+-- for every transaction, we need the following:
```

Comment:
I think documentation for this module is more relevant than for others.  `docs/concepts/AcidPoly.lhs` is a big help, but it is not very specific on the `TH` api.

I would like this module to have a haddock section, that links to the concepts and then elaborates on the `TH` specifics, and answers questions like: How do you call which function if you want to use this as a library?  (Or if you want to extend the kernel?)